src/popup/components/Modals/MultisigVaultCreate.vue

```diff
@@ -0,0 +1,330 @@
+    BtnText,
+    BtnHelp,
+    InputField,
+    MultisigVaultCreationProgress,
+    QrScanIcon,
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    reject: { type: Function, required: true },
+  },
+  setup(props, { root }) {
+    const { deployMultisigAccount, multisigProgress } = useMultisig({ store: root.$store });
+    const multisigAccount = ref<IMultisigAccount | null>(null);
+    const signers = ref<ICreateMultisigAccount[]>([]);
+    const isCreatingVault = ref<boolean>(false);
+    const requiredNumOfConfirmations = ref<number>(MULTISIG_VAULT_MIN_NUM_OF_SIGNERS);
+    const minNumOfSigners = computed(
+      () => signers.value.length < MULTISIG_VAULT_MIN_NUM_OF_SIGNERS
+        ? MULTISIG_VAULT_MIN_NUM_OF_SIGNERS
+        : signers.value.length,
+    );
+    const isValidSigners = computed(
+      () => !signers.value.filter((signer) => !validateHash(signer.address).valid).length,
+    );
+    const canCreateMultisig = computed(
+      () => signers.value.length >= MULTISIG_VAULT_MIN_NUM_OF_SIGNERS && isValidSigners.value,
+    );
+    const showVaultCreationProgress = computed<boolean>(
+      () => isCreatingVault.value || !!multisigAccount.value,
+    );
+
+    function addNewSigner() {
+      signers.value.push({
+        address: '',
+      });
+    }
+
+    /**
+     * Scans a QR code and add a signer address
+     * @param {number} signerIndex - The index of the signer to update the address of
+     */
+    async function openScanQrModal(signerIndex: number) {
+      const scanResult = await root.$store.dispatch('modals/open', {
+        name: MODAL_READ_QR_CODE,
+        title: root.$t('pages.send.scanAddress'),
+        icon: 'critical',
+      });
+
+      if (!scanResult) return;
+
+      const { valid, isName } = validateHash(scanResult);
+
+      // check if the address is valid and it's not a name
+      if (!(valid && !isName)) {
+        root.$store.dispatch('modals/open', {
+          name: MODAL_DEFAULT,
+          title: root.$t('modals.invalid-qr-code.msg'),
+          icon: 'critical',
+        });
+        return;
+      }
+
+      // Check if signer address already added
+      if (signers.value.find((signer) => signer.address === scanResult)) {
+        root.$store.dispatch('modals/open', {
+          name: MODAL_DEFAULT,
+          title: root.$t('modals.createMultisigAccount.errorDuplicatingSigner'),
+          icon: 'critical',
+        });
+        return;
+      }
+
+      // Update the signer address
+      signers.value[signerIndex].address = scanResult;
+    }
+
+    function getSignerLabel(index: number) {
+      return `${root.$t('modals.createMultisigAccount.signer')} ${index + 1}`;
+    }
+
+    async function createMultisigAccount() {
+      isCreatingVault.value = true;
+      multisigAccount.value = await deployMultisigAccount(
+        requiredNumOfConfirmations.value,
+        signers.value.map((signer) => signer.address) as [string, string, ...string[]],
+      );
```

Comment:
In my opinion this should be wrapped in try..catch block and some action should be done in case of an error.
I tested the flow on my side and it stuck after the failure on one of the steps.